### ISSUE TITLE
merge does not work in node cli

### ISSUE THREAD

**Issue by bhoriuchi** [neumino/rethinkdbdash#271]
Not a huge issue and it may just be my setup but I ran into an issue when using the merge method in in node cli (node v6.3.0). The equivalent code has no issues when run inside a .js file

here is an example to reproduce

```js
var r = require('rethinkdbdash')()

// table data
var list = [
  {
    id: '358ba8e6-838a-4f92-825d-7c25f85e42ab',
    name: 'list1',
    type: '4effeadb-19a6-4b76-9596-10d034ed47e0'
  }
]

var list_type = [
  {
    id: '4effeadb-19a6-4b76-9596-10d034ed47e0',
    name: 'shopping'
  }
]
// assume these are already loaded into rethinkdb as tables list and list_type

r.table('list').merge(function (lists) {
  return {
    type: r.table('list_type').get(lists('type'))
  }
}).run().then(function (results) {
  console.log(results)
})
```

works fine in a .js file but from cli throws

```
TypeError: Cannot read property 'unshift' of undefined
    at Term._fillArgs (/***/node_modules/rethinkdbdash/lib/term.js:2918:17)
    at Term.merge (/***/node_modules/rethinkdbdash/lib/term.js:1363:8)
    at repl:1:25
    at sigintHandlersWrap (vm.js:32:31)
    at sigintHandlersWrap (vm.js:96:12)
    at ContextifyScript.Script.runInContext (vm.js:31:12)
    at REPLServer.defaultEval (repl.js:308:29)
    at bound (domain.js:280:14)
    at REPLServer.runBound [as eval] (domain.js:293:12)
    at REPLServer.<anonymous> (repl.js:489:10)
    at emitOne (events.js:101:20)
    at REPLServer.emit (events.js:188:7)
    at REPLServer.Interface._onLine (readline.js:232:10)
    at REPLServer.Interface._line (readline.js:574:8)
    at REPLServer.Interface._ttyWrite (readline.js:851:14)
    at ReadStream.onkeypress (readline.js:119:10)

```

and when printing the args _error has

```
    _error: 
     { ReqlDriverError: Cannot convert `function (lists) { return {type: r.table('list_type').get(lists('type'))} }` to datum..
         at Term.expr (/***/node_modules/rethinkdbdash/lib/term.js:2400:21)
         at Term.merge (/***/node_modules/rethinkdbdash/lib/term.js:1361:33)
         at repl:1:61
         at sigintHandlersWrap (vm.js:32:31)
         at sigintHandlersWrap (vm.js:96:12)
         at ContextifyScript.Script.runInContext (vm.js:31:12)
         at REPLServer.defaultEval (repl.js:308:29)
         at bound (domain.js:280:14)
         at REPLServer.runBound [as eval] (domain.js:293:12)
         at REPLServer.<anonymous> (repl.js:489:10)
         at emitOne (events.js:101:20)
         at REPLServer.emit (events.js:188:7)
         at REPLServer.Interface._onLine (readline.js:232:10)
         at REPLServer.Interface._line (readline.js:574:8)
         at REPLServer.Interface._ttyWrite (readline.js:851:14)
         at ReadStream.onkeypress (readline.js:119:10)
       msg: 'Cannot convert `function (lists) { return {type: r.table(\'list_type\').get(lists(\'type\'))} }` to datum.',
       message: 'Cannot convert `function (lists) { return {type: r.table(\'list_type\').get(lists(\'type\'))} }` to datum..' }
```

**Comment by neumino** [comment]
Looked a bit into it and opened https://github.com/nodejs/node/issues/8089 to get some clarification on the expected behavior. If this is working as intended, I could convert the `x instanceof Function` to `typeof x === 'function'`.

**Comment by neumino** [comment]
There's actually also an issue with the `unshift` error. This is because the code is going in a path that should never be reached... I'll wait a bit for some answer, and will push this fix regardless of whether the issue is working as intended or not.

**Comment by neumino** [comment]
Looks like it's a bug in node CLI -- https://github.com/nodejs/node/issues/7788

**Comment by neumino** [comment]
Ok, I just pushed a fix in 2.3.20 for the `unshift` error message. The root issue is in node, not in rethinkdbdash, so I'm closing this issue.

**Comment by bhoriuchi** [comment]
great, thanks for the quick and thorough response
